server/task_check_scheduler.go

```diff
@@ -181,7 +181,8 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 // TL;DR
 // 1. We do not want anything happen if user does not set this field at the frontend.
 // 2. If user has specified this field once (even if she sets it to default latter), we will schedule one ever since.
-// 3. We only want to schedule one if the payload field is different from the last run (unless user forces one)
+// 3. We only want to schedule one if the payload field is different from the last run (unless user forces one).
+//    (This rule is always satisfied since every time when users patch the EarliestAllowedTs, we will check if this field has been modified and force scheduling one if needed.)
```

Comment:
Returns true if we meet either of the following conditions:

1.  Task has a non-default value and no task check has run before (so we are about to kick of the check the first time)
2.  The specified EarliestAllowedTs has elapsed so we need to rerun the check to unblock the task.

On the other hand, we would also rerun the check if user has modified EarliestAllowedTs. This is handled separately in the task patch handler.